hail/src/main/scala/is/hail/expr/ir/MapIR.scala

```diff
@@ -8,3 +17,27 @@ object MapIR {
 
   def mapBaseIR(ir: BaseIR, f: BaseIR => BaseIR): BaseIR = f(ir.copy(newChildren = ir.children.map(mapBaseIR(_, f))))
 }
+
+/**
+  * Would be more elegant in a mutable IR with parent pointers
+  */
+object MapIRSubtrees {
```

Comment:
I'm pretty sure this just RewriteBottomUp, no?

Reply:
It's not quite... the `f` here is one that should be applied to trees of value IRs. RewriteBottomUp matches on each BaseIR node.

Reply:
OK.  We should probably keep `BottomUp` in the name, since top-down is also reasonable.  What about `RewriteIRBottomUp`?